src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -200,17 +200,24 @@ private static FilePath tempDir(FilePath ws) {
         private final String container;
         private final String[] envHost;
         private final String ws;
+        private final String userId;
 
-        Decorator(String container, EnvVars envHost, String ws) {
+        Decorator(String container, EnvVars envHost, String ws, String userId) {
             this.container = container;
             this.envHost = Util.mapToEnv(envHost);
             this.ws = ws;
+            this.userId = userId;
         }
 
         @Override public Launcher decorate(final Launcher launcher, Node node) {
             return new Launcher.DecoratedLauncher(launcher) {
                 @Override public Proc launch(Launcher.ProcStarter starter) throws IOException {
-                    List<String> prefix = new ArrayList<String>(Arrays.asList("docker", "exec", container, "env"));
+                    List<String> prefix;
+                    if (userId == null) {
+                        prefix = new ArrayList<String>(Arrays.asList("docker", "exec", container, "env"));
```

Comment:
Good, this is necessary for compatibility with builds running before the upgrade.